spyder/workers/updates.py

```diff
@@ -95,5 +96,7 @@ def start(self):
         except Exception:
             error_msg = _('Unable to check for updates.')
 
-        self.error = error_msg
-        self.sig_ready.emit()
+        # Don't show dialog when starting up spyder and an error occur
+        if not self.startup or error_msg is None:
```

Comment:
Ok, but if `error_msg` is *different* from `None`, we also want to show the dialog. And that's not the case here because we enter this block if `error_msg` is `None` (which is a contradiction for me and what motivated me to write my first comment).